kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -131,26 +133,26 @@ evalKEq true _ _ _ (valid :< app) =
     false = not true
     sort = Attribute.patternSort valid
     Application { applicationChildren } = app
-    evalEq t1 t2 = do
-        let expr = Or.simplifyEvaluated
-                (OrPattern.fromPattern
-                    (Conditional
-                        (Bool.asInternal sort true)
-                        (Predicate.makeEqualsPredicate t1 t2)
-                        mempty
-                    )
-                )
-                (OrPattern.fromPattern
-                    (Conditional
-                        (Bool.asInternal sort false)
-                        ( Predicate.makeNotPredicate $
-                            Predicate.makeEqualsPredicate t1 t2
-                        )
-                        mempty
-                    )
-                )
+    evalEq termLike1 termLike2 = do
+        let pattern1 = Pattern.fromTermLike termLike1
+            pattern2 = Pattern.fromTermLike termLike2
+
+        defined1 <- Ceil.makeEvaluate Predicate.topTODO pattern1
+        defined2 <- Ceil.makeEvaluate Predicate.topTODO pattern2
+        defined <- And.simplifyEvaluated defined1 defined2
+
+        equalTerms <- Equals.makeEvaluate pattern1 pattern2 Predicate.topTODO
```

Comment:
makeEvaluateTermsToPredicate, maybe? If that works, you might be able to remove the replaceTerm stuff, maybe also the fromTermLike stuff above.

Reply:
I'm going to split this part of the pull request off into its own change set because it's not related to this particular bug, but something else I found while looking for this one.